### PR TITLE
CIRC-1375 TLR response does not contain instance data

### PR DESCRIPTION
**Pull request by bisecomsergiy**
### Purpose
Get TLR response with necessary data of instance
### Approach
Added get methods to the instance class to get necessary values of instanceRepresentation;
Updated addInstanceProperties accordingly
### Learning
https://issues.folio.org/browse/CIRC-1375

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java [hunk 1]
```diff
@@ -175,10 +182,15 @@ void get(RoutingContext routingContext) {
...
-
-    fromFutureResult(requestRepository.getById(id))
+    final var instanceRepository = new InstanceRepository(clients);
+    final var permissions = OkapiPermissions.from(context.getHeaders());
+    final var errorHandler = new OverridingErrorHandler(permissions);
+
+    fromFutureResult(requestRepository.getById(id)
+      .thenComposeAsync(r -> r.after(when(
+        req -> shouldFetchInstance(errorHandler, req),
+        req -> fetchInstance(instanceRepository, req), req -> ofAsync(() -> req)))))
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/RequestCollectionResource.java:193]: Should this be included in requestRepository.getById() ?

File: src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java [hunk 2]
```diff
@@ -38,15 +39,17 @@ private static void addStoredItemProperties(JsonObject request, Item item) {
     request.put("item", itemSummary);
   }
 
-  private static void addStoredInstanceProperties(JsonObject request, Item item) {
-    if (item == null || item.isNotFound()) {
-      logUnableAddItemToTheRequest(request, item);
+  private static void addStoredInstanceProperties(JsonObject request, Instance instance) {
+    if (Objects.isNull(instance) || instance.isNotFound()) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java:43]: Why not just '== null' ?

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java [hunk 3]
```diff
@@ -175,10 +182,15 @@ void get(RoutingContext routingContext) {
     final var clients = Clients.create(context, client);
 
     final var requestRepository = RequestRepository.using(clients);
-
     final var id = getRequestId(routingContext);
-
-    fromFutureResult(requestRepository.getById(id))
+    final var instanceRepository = new InstanceRepository(clients);
+    final var permissions = OkapiPermissions.from(context.getHeaders());
+    final var errorHandler = new OverridingErrorHandler(permissions);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/RequestCollectionResource.java:188]: Why OverridingErrorHandler?

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java (hunk 1 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/resources/RequestCollectionResource.java]: Moved fetch to mentioned method, thank you.

File: src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java (hunk 2 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java:43]: Will follow the style, thanks!

Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/domain/StoredRequestRepresentation.java:43]: And it is done

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java (hunk 3 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/resources/RequestCollectionResource.java]: Moved the handler, thnaks

File: src/main/java/org/folio/circulation/domain/RequestRepresentation.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import static org.folio.circulation.support.utils.DateFormatUtil.formatDateTime;
 
 import java.lang.invoke.MethodHandles;
+import java.util.Objects;
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/RequestRepresentation.java:13]: Unused import?

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java [hunk 5]
```diff
@@ -308,4 +315,13 @@ private RequestNoticeSender createRequestNoticeSender(Clients clients,
...
   }
+
+  private CompletableFuture<Result<Request>> fetchInstance(InstanceRepository instanceRepository, Request request) {
+    return succeeded(request)
+      .combineAfter(instanceRepository::fetch, Request::withInstance);
+  }
+
+  private CompletableFuture<Result<Boolean>> shouldFetchInstance(CirculationErrorHandler errorHandler, Request request) {
+    return ofAsync(() -> errorHandler.hasNone(INVALID_INSTANCE_ID));
+  }
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/RequestCollectionResource.java:326]: Unused methods?

File: src/main/java/org/folio/circulation/domain/RequestRepresentation.java (hunk 4 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/domain/RequestRepresentation.java]: Sorry, it is moved

File: src/main/java/org/folio/circulation/resources/RequestCollectionResource.java (hunk 5 above)
Comment (bisecomsergiy) [reply src/main/java/org/folio/circulation/resources/RequestCollectionResource.java]: Sorry, it is moved..

**Review by alexanderkurash — APPROVED** [review]

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**